x-pack/plugins/security_solution/scripts/assistant/attack_discovery/load.ts

```diff
@@ -0,0 +1,327 @@
+  let indexName: string;
+
+  if (indexType === 'alerts') {
+    indexName = `insights-alerts-ep${epNum}-${indexNameSuffix}`;
+  } else {
+    indexName = `logs-endpoint.events.insights.ep${epNum}.${indexNameSuffix}`;
+  }
+
+  log.info(`Processing and indexing file: ${file} ...`);
+
+  const fileData = await fs.readFileSync(file).toString().split('\n');
+
+  try {
+    const response = await esClient.bulk<string>({
+      pipeline: PIPELINE_NAME,
+      pretty: true,
+      body: [
+        fileData.reduce((acc, item) => {
+          if (!item.length) return acc;
+
+          return acc.concat(`{ "index" : { "_index" : "${indexName}" } }\n${item}\n`);
+        }, ''),
+      ],
+    });
+    if (!response.errors) {
+      log.info('Success.');
+    } else {
+      log.info(`Failed with errors.`);
+    }
+  } catch (error) {
+    log.error('Error indexing data:');
+    log.error(error);
+  }
+};
+
+const processFilesForEpisode = async ({
+  esClient,
+  epNum,
+  log,
+}: {
+  esClient: Client;
+  epNum: string;
+  log: ToolingLog;
+}) => {
+  const dataFiles = fs
+    .readdirSync(DIRECTORY_PATH)
+    .filter((file) => file.includes(`ep${epNum}data.ndjson`));
+  const alertFiles = fs
+    .readdirSync(DIRECTORY_PATH)
+    .filter((file) => file.includes(`ep${epNum}alerts.ndjson`));
+
+  for (const file of dataFiles) {
+    await processFile({ esClient, file: path.join(DIRECTORY_PATH, file), indexType: 'data', log });
+  }
+
+  for (const file of alertFiles) {
+    await processFile({
+      esClient,
+      file: path.join(DIRECTORY_PATH, file),
+      indexType: 'alerts',
+      log,
+    });
+  }
+};
+
+const checkRuleExistsAndStatus = async ({
+  kbnClient,
+  log,
+}: {
+  kbnClient: KbnClient;
+  log: ToolingLog;
+}) => {
+  log.info("Checking if the rule 'Endpoint Security [Insights]' exists and its status...");
+
+  try {
+    const rule = await getRule({ kbnClient, log });
+
+    if (!rule) {
+      await importRule({ kbnClient, log });
+    } else if (!rule.enabled) {
+      await enableRule({ kbnClient, ruleId: rule.id, log });
+    }
+  } catch (error) {
+    log.error('Error checking rule status:');
+    log.error(error);
+  }
+};
+
+const checkDeleteIndices = async ({ esClient, log }: { esClient: Client; log: ToolingLog }) => {
+  const promptDeletion = async (pattern: string, description: string) => {
+    try {
+      const response = await esClient.cat.indices({
+        index: [pattern],
+        h: 'index',
+      });
+
+      const existingIndices = (response as unknown as string).trim().split(' ');
+
+      if (existingIndices.length > 0) {
+        log.info(
+          `Found existing ${description} indices matching pattern '${pattern}': ${existingIndices.join(
+            ', '
+          )}`
+        );
+
+        // TODO: add deletion logic
+        // for (const index of existingIndices) {
+        //   await esClient.indices.delete({ index: pattern });
+
+        //   log.info(`Deleted index ${index}`);
+        // }
```

Comment:
Why is the index delete logic commented out?

Reply:
it was in the original script we have used locally, but I wasn't sure if it's something we will need in our tests, but just wanted to make sure we don't loose it

Reply:
okay, if you know the author of the original script it would be great to find out why this was commented out. 
If this is needed we should uncomment it, and if not there's no point in having this `checkDeleteIndices` function implemented without executing the actual deletion, right now it's called only to log stuff without doing anything.
Just a suggestion to keep the code clean.